terra/projects/permissions.py

```diff
@@ -128,3 +129,9 @@ class IsModelVersionModelPublic(permissions.BasePermission):
 
     def has_object_permission(self, request, view, obj):
         return obj.model.is_public
+
+class HasBetaAccess(permissions.BasePermission):
+    message = 'This feature is only available to beta users.'
+
+    def has_permission(self, request, view):
```

Comment:
I had to use `has_permission` instead of `has_object_permission`, for some reason it wasn't being called and didn't work.